WD-Firmware/psp/src/psp_interrupts.c

```diff
@@ -0,0 +1,295 @@
+/* 
+* SPDX-License-Identifier: Apache-2.0
+* Copyright 2019 Western Digital Corporation or its affiliates.
+* 
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* 
+* http:*www.apache.org/licenses/LICENSE-2.0
+* 
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/**
+* @file   psp_interrupts.c
+* @author Ronen Haen
+* @date   20.05.2019
+* @brief  The file supllies interrups services such as  - registration, default handlers and indication
+*         whether we are in interrpu context
+* 
+*/
+
+/**
+* include files
+*/
+#include "psp_types.h"
+#include "psp_api.h"
+
+/**
+* definitions
+*/
+
+/* The stack used by interrupt service routines */
+#ifdef D_ISR_STACK_SIZE_WORDS
+	/*D_PSP_DATA_SECTION*/ static D_PSP_16_ALIGNED pspStack_t xISRStack[ D_ISR_STACK_SIZE_WORDS ] = { 0 };
+	const pspStack_t xISRStackTop = ( pspStack_t ) &( xISRStack[ ( D_ISR_STACK_SIZE_WORDS ) - 1 ] );
+#else
+    #error "ISR Stack size is not defined"
+#endif
+
+
+/* Set configCHECK_FOR_STACK_OVERFLOW to 3 to add ISR stack checking to task
+stack checking.  A problem in the ISR stack will trigger an assert, not call the
+stack overflow hook function (because the stack overflow hook is specific to a
+task stack, not the ISR stack). */
+#if( D_CHECK_FOR_STACK_OVERFLOW > 2 )
+	#warning This path not tested, or even compiled yet.
+	/* Don't use 0xa5 as the stack fill bytes as that is used by the kernerl for
+	the task stacks, and so will legitimately appear in many positions within
+	the ISR stack. */
+	#define D_PSP_ISR_STACK_FILL_BYTE	0xee
+
+	static const u08_t ucExpectedStackBytes[] = {
+									D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE,		\
+									D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE,		\
+									D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE,		\
+									D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE,		\
+									D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE, D_PSP_ISR_STACK_FILL_BYTE };	\
+
+	#define M_PSP_CHECK_ISR_STACK() configASSERT( ( memcmp( ( void * ) xISRStack, ( void * ) ucExpectedStackBytes, sizeof( ucExpectedStackBytes ) ) == 0 ) )
+#else
+	/* Define the function away. */
+	#define M_PSP_CHECK_ISR_STACK()
+#endif /* D_CHECK_FOR_STACK_OVERFLOW > 2 */
+
+
+/**
+* macros
+*/
+
+/**
+* types
+*/
+u32_t g_uiIsInterruptContext = D_PSP_NON_INT_CONTEXT;
```

Comment:
need to go to rtosal_util (with the other hooks)

Reply:
done https://github.com/westerndigitalcorporation/riscv-fw-infrastructure/pull/12/commits/6c26f8aba68f07621c53b564702cbab8a7d91dd1